keel-web/src/main/kotlin/com/netflix/spinnaker/keel/apidocs/DurationPropertyCustomizer.kt

```diff
@@ -0,0 +1,22 @@
+package com.netflix.spinnaker.keel.apidocs
+
+import io.swagger.v3.core.converter.AnnotatedType
+import io.swagger.v3.oas.models.media.Schema
+import java.time.Duration
+import org.springdoc.core.customizers.PropertyCustomizer
+import org.springframework.stereotype.Component
+
+@Component
+class DurationPropertyCustomizer : PropertyCustomizer {
+
+  override fun customize(property: Schema<*>?, type: AnnotatedType): Schema<*>? =
+    property?.apply {
+      if (type.baseType == Duration::class.java) {
+        // OpenAPI ignores the return value so we have to just modify the schema in place.
+        // See https://github.com/springdoc/springdoc-openapi/issues/441
+        type("string")
+        format("duration")
+        properties = emptyMap()
+      }
+    }
```

Comment:
Raised  springdoc/springdoc-openapi#441 because the way property customizers work is a little weird but there's a workaround.